src/SecretManager/Microsoft.DncEng.SecretManager/Commands/ProjectBaseCommand.cs

```diff
@@ -0,0 +1,89 @@
+﻿
+using System;
+using Microsoft.DncEng.CommandLineLib;
+using Mono.Options;
+
+namespace Microsoft.DncEng.SecretManager.Commands
+{
+    /// <summary>
+    /// This class is used to extend the CommandLineLib.GlobalCommand class to add a global options spacific to this project
+    /// </summary>
+    public class ProjectBaseCommand : GlobalCommand
+    {
+
+        /// <summary>
+        /// Indicates if the global option for 'quiet' is set
+        /// </summary>
+        public bool Quiet { get { return Verbosity == VerbosityLevel.Quiet; } }
+
+        /// <summary>
+        /// Check for local environment values to indicate you are running for Azure DevOps
+        /// SYSTEM_COLLECTIONURI is a default environment variable in Azure DevOps
+        /// </summary>
+        private bool RunningInAzureDevOps = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("SYSTEM_COLLECTIONURI"));
+
+        /// <summary>
+        /// Provides the ServiceTreeId set with global options
+        /// The ID is a guid and is set to Guid.Empty if not set
+        /// </summary>
+        public Guid ServiceTreeId = Guid.Empty;
+
+        /// <summary>
+        /// Base constructor for the ProjectBaseCommand class
+        /// </summary>
+        public ProjectBaseCommand(GlobalCommand globalCommand)
+        {
+            Verbosity = globalCommand.Verbosity;
+            Help = globalCommand.Help;
+        }
+
+        /// <summary>
+        /// Overrides the GetOptions method from the base class to add a custom option for the ServiceTreeId
+        /// </summary>
+        public override OptionSet GetOptions()
+        {
+            return new OptionSet()
+            {
+                {"servicetreeid=", "The service tree ID (must be a valid GUID id from aka.ms/servicetree)", id =>
```

Comment:
what enforces validity in Service Tree❓

Reply:
Nothing other than it being a 'Guid'. The audit logging process from does not verify that a ID is valid just that it is passed and the value meets the data type criteria.

If you pass an empty Guid or a valid Guid for a non-existent service tree ID it would still 'log' to Geneva successfully.

I don't think we have an additional obligation to make this process have to communicate with an external Service Tree API to ensure IDs are valid. 

If the 'service' does not use a correct ID then it will still report to Geneva but Geneva will not close any SFI KPI work items associated with the service so in that sense you are incentivized to use a valid and correct ID.

Reply:
I do not think there is any reasonable way for this library to ensure it is a valid service tree ID. I think about this instead as being sort of "tagging" information or metadata that gets passed to Geneva, then Geneva happens to do something interesting with it.

Reply:
I'm fine w/ the level of validation implemented here. my main concern is the option's description implies something more. how about the following❓
```suggestion
                {"servicetreeid=", "The service tree ID (must be a non-empty GUID and should be a valid GUID id from https://aka.ms/servicetree)", id =>
```

Reply:
I get what you mean about the description for input but 'should be' implies that something else is OK to use here.

It is not. The value is only meaningful if it is a valid service tree ID and specifically the service tree ID for your service. 

Any other Guid even one for a valid service tree ID for a service other than the one running the process would be invalid.

The value 'would' be accepted by Geneva and recorded but it is not correct and does not benefit the actual service that needs to report so it can close it's KPIs.

However, the way this option is being provided allows for service we do not own to also use this tool and generate a log that can go to Geneva.

We cannot predict what those possible IDs would be or how they will change in the future.

What we do know is that the ID is meant to be a Guid structure that comes from https://aka.ms/servicetree and relates to the specific service owned by the 'caller'.

I will change the comment to be something less directive so it does state must or should and instead just provides the link back to the location where valid IDs can be found.

Reply:
Nothing enforces the use of a 'real' service tree id including the tools provided by the Geneiva auditing team. Since they don't require a 'real' id we do not require a 'real' id. We do however require that the ID be the same data type that is expected for service tree IDs.